daemon/wrapnix.go

```diff
@@ -0,0 +1,68 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"fmt"
+	"net"
+	sys "syscall"
+)
+
+type wrapnixAddr struct {
+	net.UnixAddr
+	ucred *sys.Ucred
+}
+
+func (wa *wrapnixAddr) String() string {
+	return fmt.Sprintf("%d:%s", wa.ucred.Uid, wa.UnixAddr.String())
+}
+
+type wrapnixConn struct {
+	net.UnixConn
```

Comment:
true story: i've been working in go for ... how long now? and i hadn't picked up on the fact that you could embed pointers